codegen/codegen_test.go

```diff
@@ -905,3 +905,57 @@ func TestCodeGen(t *testing.T) {
 		})
 	}
 }
+
+func TestCodegenError(t *testing.T) {
+	t.Parallel()
+
+	type errorTestCase struct {
+		name          string
+		targets       []string
+		input         string
+		expectedError string
+	}
+
+	for _, tc := range []errorTestCase{
+		{
+			"invalid builtin",
+			[]string{"default"},
+			`
+			fs default() {
+				bug includePatterns("*.go")
+			}
+			fs bug(option::local pattern) {
+				local "." with pattern
+			}
+		`,
+			"<stdin>:3:9: unrecognized builtin `includePatterns`",
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := diagnostic.WithSources(context.Background(), builtin.Sources())
+			mod, err := parser.Parse(context.Background(), strings.NewReader(tc.input))
```

Comment:
This should also use `ctx` from above. Sources parsed by `parser.Parse` will added via the `ctx`.

Reply:
Thanks! - fixed